### PR TITLE
feat: onboarding intercom v2 retl support

### PR DESCRIPTION
**Pull request by manish339k**
## What are the changes introduced in this PR?

Onboarding Intercom V2 rETL support on VDM V2. Transforming Record event.

## What is the related Linear task?

Resolves INT-2742

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

@coderabbitai review

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added in **new readability format**?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3843/test-report.html

**Review by ItsSudip — APPROVED** [review]

**Comment by sandeepdsvs** [comment]
@manish339k Can you add some tests covering the uncovered lines?

**Review by sandeepdsvs — APPROVED** [review]

**Review by ItsSudip — APPROVED** [review]

File: src/v0/destinations/intercom_v2/utils.js [hunk 1]
```diff
@@ -99,10 +101,18 @@ const getResponse = (method, endpoint, headers, payload) => {
...
+  let lookupFieldValue = null;
+  if (messageType === EventType.RECORD) {
+    const { identifiers } = message;
+    [[lookupField, lookupFieldValue]] = Object.entries(identifiers);
+  } else {
+    lookupField = getLookUpField(message);
+    lookupFieldValue = getFieldValueFromMessage(message, lookupField);
+    if (!lookupFieldValue) {
+      lookupFieldValue = message?.context?.traits?.[lookupField];
+    }
```
Comment (vinayteki95) [inline src/v0/destinations/intercom_v2/utils.js:115]: Just a suggestion to make it more readable:
You can probably encapsulate the if and else statement into a function inside itself for more readability and separation of message parsing which is not related to seachContact.

```
const { message, destination, metadata } = event;
  const messageType = getEventType(message);

  const extractFieldAndValue = (message, messageType) => {
    if (messageType === EventType.RECORD) {
      const { identifiers } = message;
      return Object.entries(identifiers)[0];
    }
    const lookupField = getLookUpField(message);
    let lookupFieldValue = getFieldValueFromMessage(message, lookupField) || 
                           message?.context?.traits?.[lookupField];
    return [lookupField, lookupFieldValue];
  };

  const [lookupField, lookupFieldValue] = extractFieldAndValue(message, messageType);

  if (!lookupField || !lookupFieldValue) {
    throw new Error('Missing lookup field or value for searchContact');
  }
```

**Review by vinayteki95 — APPROVED** [review]

File: src/v0/destinations/intercom_v2/utils.js (hunk 1 above)
Comment (manish339k) [reply src/v0/destinations/intercom_v2/utils.js]: done.

**Review by koladilip — APPROVED** [review]

**Pull request merged**